src/postgres/commands/PostgresQueryWizard/FunctionQuerySteps/FunctionQueryReturnTypeStep.ts

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { QuickPickItem } from "vscode";
+import { AzureWizardPromptStep, IWizardOptions } from "vscode-azureextensionui";
+import { ext } from "../../../../extensionVariables";
+import { localize } from "../../../../utils/localize";
+import { FunctionQueryCustomReturnTypeStep } from "./FunctionQueryCustomReturnTypeStep";
+import { IPostgresFunctionQueryWizardContext } from "./IPostgresFunctionQueryWizardContext";
+
+const customReturnTypePick: QuickPickItem = { label: localize('enterCustomReturnType', '$(pencil) Enter custom return type...') };
+
+export class FunctionQueryReturnTypeStep extends AzureWizardPromptStep<IPostgresFunctionQueryWizardContext> {
+    public async prompt(wizardContext: IPostgresFunctionQueryWizardContext): Promise<void> {
+        const returnTypeQuickPicks: QuickPickItem[] = returnTypes.map(r => { return { label: r }; });
+        returnTypeQuickPicks.push(customReturnTypePick);
+
+        wizardContext.returnTypePick = (await ext.ui.showQuickPick(returnTypeQuickPicks, {
+            placeHolder: localize('selectReturnType', 'Select return type')
+        }));
+    }
+
+    public shouldPrompt(): boolean {
+        return true;
+    }
+
+    public async getSubWizard(wizardContext: IPostgresFunctionQueryWizardContext): Promise<IWizardOptions<IPostgresFunctionQueryWizardContext> | undefined> {
+        if (wizardContext.returnTypePick === customReturnTypePick) {
```

Comment:
So it's a little weird to me to have `returnTypePick` on the wizard context. I would leverage `IAzureQuickPickItem`, specifically the `data` field.

The picks could be defined like this:
```
const returnTypeQuickPicks: IAzureQuickPickItem<string | undefined>[] = returnTypes.map(r => { return { label: r, data: r }; });
returnTypeQuickPicks.push({ label: localize('enterCustomReturnType', '$(pencil) Enter custom return type...'), data: undefined });
```

You add just `data` to the wizardContext (not the pick) and then here you just do this:
```
return wizardContext.returnType ? undefined : { promptSteps: [new FunctionQueryCustomReturnTypeStep()] };
```